src/components/productsGrid/ProductsGrid.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  Button,
+  LoadingMessage,
+  ErrorMessage,
+  HasNoResultsMessage,
+} from "../ui";
+import { ProductCard } from "./ProductCard";
+import { useProducts } from "../../hooks/useProducts";
+import { Product } from "../../types";
+import { useState } from "react";
+
+type ProductsGridProps = {
+  searchTerm: string;
+  sortBy: string;
+  sortOrder: string;
+};
+
+export const ProductsGrid = ({
+  searchTerm,
+  sortBy,
+  sortOrder,
+}: ProductsGridProps) => {
+  const [limit, setLimit] = useState(6);
+  const [isLoadingMore, setIsLoadingMore] = useState(false);
+  const {
+    data = [],
+    isLoading,
```

Comment:
`isFetching` is more flexible because it covers both the initial load and any background refetches, even when cached data is available.
`isLoading` is only true on the very first load, so it’s less useful when data is being refetched in the background, so use only if you want to consider that scenario